demo-apps/cf-helloworld-client/src/main/java/org/eclipse/californium/examples/MulticastTestClient.java

```diff
@@ -0,0 +1,117 @@
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Bosch Software Innovations - initial creation
+ ******************************************************************************/
+package org.eclipse.californium.examples;
+
+import java.io.File;
+
+import org.eclipse.californium.core.CoapClient;
+import org.eclipse.californium.core.CoapHandler;
+import org.eclipse.californium.core.CoapResponse;
+import org.eclipse.californium.core.Utils;
+import org.eclipse.californium.core.coap.CoAP;
+import org.eclipse.californium.core.coap.CoAP.Type;
+import org.eclipse.californium.core.coap.Request;
+import org.eclipse.californium.core.network.CoapEndpoint;
+import org.eclipse.californium.core.network.config.NetworkConfig;
+import org.eclipse.californium.core.network.config.NetworkConfig.Keys;
+import org.eclipse.californium.core.network.config.NetworkConfigDefaultHandler;
+
+public class MulticastTestClient {
+
+	/**
+	 * File name for network configuration.
+	 */
+	private static final File CONFIG_FILE = new File("CaliforniumMulticast.properties");
+	/**
+	 * Header for network configuration.
+	 */
+
+	private static final String CONFIG_HEADER = "Californium CoAP Properties file for Multicast Client";
+	/**
+	 * Special network configuration defaults handler.
+	 */
+	private static NetworkConfigDefaultHandler DEFAULTS = new NetworkConfigDefaultHandler() {
+
+		@Override
+		public void applyDefaults(NetworkConfig config) {
+			config.setInt(Keys.MULTICAST_BASE_MID, 65000);
+		}
+
+	};
+
+	public static void main(String args[]) {
+
+		NetworkConfig config = NetworkConfig.createWithFile(CONFIG_FILE, CONFIG_HEADER, DEFAULTS);
+
+		CoapEndpoint endpoint = new CoapEndpoint.CoapEndpointBuilder().setNetworkConfig(config).build();
+		CoapClient client = new CoapClient();
+
+		client.setURI("coap://localhost:5683/helloWorld");
+		client.setEndpoint(endpoint);
+		Request request = Request.newGet();
+		request.setType(Type.CON);
+		client.setEndpoint(endpoint);
+
+		// sends an uni-cast request
+		CoapResponse response = client.advanced(request);
+		if (response != null) {
+			System.out.println(Utils.prettyPrint(response));
+		} else {
+			System.out.println("No response received.");
+		}
+
+		client.setURI("coap://" + CoAP.MULTICAST_IPV4.getHostAddress() + "/helloWorld");
+		Request multicastRequest = Request.newGet();
+		multicastRequest.setType(Type.NON);
+		// sends a multicast request
+		client.advanced(handler, multicastRequest);
+		while (handler.waitOn(2000))
+			;
+	}
+
+	private static final MultiCoapHandler handler = new MultiCoapHandler();
+
+	private static class MultiCoapHandler implements CoapHandler {
```

Comment:
using synchronized onWait() and on() methods looks complicated, a CountDownLatch or a sleep thread of 2000ms can be simple for a demonstration

Reply:
:-).

The idea is to wait until more then 2s no new response arrives. 
synchronize is not complex, especially, if the other primitives just have other functionality.